modules/core/src/matrix_transform.cpp

```diff
@@ -812,6 +815,53 @@ void flip( InputArray _src, OutputArray _dst, int flip_mode )
         flipHoriz( dst.ptr(), dst.step, dst.ptr(), dst.step, dst.size(), esz );
 }
 
+static void
+flipNDImpl(uchar* data, const int* shape, const size_t* step, int axis)
+{
+    int total = 1;
+    for (int i = 0; i < axis; ++i)
+        total *= shape[i];
+
+    int shape_at_axis = shape[axis];
+    size_t step_at_axis = step[axis];
+    size_t offset = 0;
+    size_t offset_increment = axis == 0 ? 0 : step[axis - 1];
+    for (int i = 0; i < total; ++i, offset += offset_increment)
+        for (int j = 0, k = shape_at_axis - 1; j < (shape_at_axis + 1) / 2; ++j, --k)
+            std::swap_ranges(data + offset + j * step_at_axis,
+                             data + offset + j * step_at_axis + step_at_axis,
+                             data + offset + k * step_at_axis);
+}
+
+void flipND(InputArray _src, OutputArray _dst, int _axis)
+{
+    CV_INSTRUMENT_REGION();
+
+    Mat src = _src.getMat();
+    CV_Assert(src.isContinuous());
+    int ndim = src.dims;
+    const auto shape = src.size.p;
+    const auto step = src.step.p;
+
+    // verify axis
+    CV_CheckLT(_axis, ndim, "flipND: given axis is out of range");
+    CV_CheckGT(_axis, -ndim, "flipND: given axis is out of range");
+    int axis = (_axis + ndim) % ndim;
+
+    // return the src if it has only one element on the flip axis
+    if (shape[axis] == 1)
```

Comment:
We don't need this special case, I think.

Reply:
It serves like an early stop to improve efficiency, doesn't it?

Reply:
It serves like an early stop to improve efficiency, doesn't it?

Reply:
To be honest, I don't know what we are winning here - one function frame that will do exactly this? I think copying is much slower than calling a function. @alalek can you elaborate?

Reply:
> I think copying is much slower than calling a function

function itself is heavier than simple copying.

---

There is no problem with that part. Refer to `convertTo()` implementation.